### PR TITLE
Add favorite registration micro-interaction for Android

### LINKED ISSUE #203: Favorite registration micro-interaction for Android
## Overview
Implement a micro-interaction animation when tapping the favorite button on the timetable list. The animation should show a heart entering the "Favorites" tab (heart icon) in the bottom navigation from the timetable list's favorite button.

## Links
- Design: https://www.figma.com/design/XUk8WMbKCeIdWD5cz9P9JC/DroidKaigi-2024-App-UI

### PR DESCRIPTION
**Pull request by shunm-999**
## Issue
- close #203 

## Overview (Required)
- Implemented a micro-interaction animation when tapping the favorite button on the timetable list. The animation show a heart falling from the FavoriteIcon in the timetable list to the heart icon in the BottomNavigation. 

## Links
- 

## Movie (Optional)
Before | After
:--: | :--:
<video src="https://github.com/user-attachments/assets/c969d0d6-10fe-4082-8da9-da61c140c1cd" width="300" > | <video src="https://github.com/user-attachments/assets/50b4c996-9cd3-496f-88d2-fd05cfb403e1" width="300" >

### CODE REVIEW

File: core/ui/src/commonMain/kotlin/io/github/droidkaigi/confsched/ui/animation/FavoriteAnimation.kt [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+            content()
+            FavoriteAnimationFrame()
+        }
+    }
+}
+
+@Composable
+private fun FavoriteAnimationFrame() {
+    val animationScope = LocalFavoriteAnimationScope.current
+    Box(
```
Comment (takahirom) [inline core/ui/src/commonMain/kotlin/io/github/droidkaigi/confsched/ui/animation/FavoriteAnimation.kt:53]: Thanks. Now I think this is using recomposition for animation. Could we use drawWithContent{} and draw phase for animation?

Comment (shunm-999) [reply core/ui/src/commonMain/kotlin/io/github/droidkaigi/confsched/ui/animation/FavoriteAnimation.kt:53]: Ok, I will try!

Comment (shunm-999) [reply core/ui/src/commonMain/kotlin/io/github/droidkaigi/confsched/ui/animation/FavoriteAnimation.kt]: @takahirom 
I have fixed it [837bd98](https://github.com/DroidKaigi/conference-app-2024/pull/536/commits/837bd9837c700c4a6d282e2289a5196de673a2ee), so please check it!

**Review by takahirom — APPROVED** [review]
Efficient implementation! Thank you for your contribution!

**Pull request merged**